sdk/mongocluster/arm-mongocluster/review/arm-mongocluster.api.md

```diff
@@ -432,6 +507,18 @@ export interface PrivateLinksOperations {
     listByMongoCluster: (resourceGroupName: string, mongoClusterName: string, options?: PrivateLinksListByMongoClusterOptionalParams) => PagedAsyncIterableIterator<PrivateLinkResource>;
 }
 
+// @public
+export type PromoteMode = string;
+
+// @public
+export type PromoteOption = string;
+
+// @public
+export interface PromoteReplicaRequest {
+    mode?: PromoteMode;
+    promoteOption: PromoteOption;
+}
+
```

Comment:
Could you share the typespec link for ResourceProvisioningState and ProvisioningState with me?

Reply:
https://github.com/kazrael2119/azure-rest-api-specs/blob/c3474b5f97987003f73a08765b3450f2829eb6ea/specification/mongocluster/DocumentDB.MongoCluster.Management/common.tsp#L12-L26

Reply:
ResourceProvisioningState :
```
/**
 * Standard terminal provisioning state of resource type. You can include in your
 * custom provision state to avoid duplication and ensure consistency
 *
 * @example
 *
 * ```typespec
 * union FooProvisioningState {
 *   ResourceProvisioningState,  // include standard provisioning states
 *   starting: "starting",
 *   started: "started",
 *   stopping: "stopping",
 *   stopped: "stopped",
 * }
 * ```
 */
@doc("The provisioning state of a resource type.")
@Azure.Core.lroStatus
union ResourceProvisioningState {
  @doc("Resource has been created.")
  Succeeded: "Succeeded",

  @doc("Resource creation failed.")
  Failed: "Failed",

  @doc("Resource creation was canceled.")
  Canceled: "Canceled",

  string,
}
```

Reply:
Should generate as 

```

export type ProvisioningState = string;

export enum KnownProvisioningState  {
    Canceled = "Canceled",
    Failed = "Failed",
    Succeeded = "Succeeded",
   InProgress = "InProgress",
   Updating = "Updating",
   Dropping = "Dropping"
}

``